Lesson17/main.js

```diff
@@ -0,0 +1,132 @@
+  const json = await fetchErrorHandling(response);
+  return json;
+}
+
+const getSlideImageData = async () => {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(getJsonOrError("https://myjson.dit.upm.es/api/bins/gzcn")), 3000);
+  })
+}
+
+const tryGetSlideImageData = async () => {
+  addLoading(slideWrapper);
+  try {
+    return await getSlideImageData();
+  } catch (e) {
+    addErrorMessage(e, slideWrapper);
+  } finally {
+    removeLoading();
+  }
+}
+
+const init = async () => {
+  const imageData = await tryGetSlideImageData();
+  if(imageData) {
+    initSlideItem(imageData); 
+    initArrowButtons();
+  }
+}
+
+const initSlideItem = (imageSrc) => {
+  for (let i = 0; i < imageSrc.length; i++) {
+    const createdSlideItem = createSlideItem(imageSrc[i]);
+    //初期設定として 最初の要素にis-displayingクラスを付与
+    i === 0 && createdSlideItem.classList.add("is-displaying");
+    addElement(createdSlideItem, slideList);
+  }
+}
+
+const initArrowButtons = () => {
+  const createdArrowButtons = createArrowButtons();
+  addElement(createdArrowButtons, slideWrapper);
+  setClickEventInArrowButton();
+}
+
+const createSlideItem = ({ image }) => {
+  const slideItem = createElementWithClassName("li", "slide-item");
+  const slideImage = createElementWithClassName("img", "slide-img");
+  slideImage.src = image;
+  slideItem.appendChild(slideImage);
+  return slideItem;
+}
+
+const addElement = (element, parent) => parent.appendChild(element);
+
+const createArrowButtons = () => {
+  const arrowBtnWrapper = createElementWithClassName("div", "arrow-btn__wrapper");
+  const prevButton = createElementWithClassName("button", "arrow-btn --prev");
+  const nextButton = createElementWithClassName("button", "arrow-btn --next");
+  prevButton.appendChild(document.createElement("span"));
+  nextButton.appendChild(document.createElement("span"));
+  prevButton.id = "js-arrow-left-btn";
+  nextButton.id = "js-arrow-right-btn";
+  arrowBtnWrapper.appendChild(prevButton);
+  arrowBtnWrapper.appendChild(nextButton);
+  return arrowBtnWrapper;
+}
+
+const setClickEventInArrowButton = () => {
+  const arrowButtons = document.querySelectorAll(".arrow-btn");
+  arrowButtons.forEach(button => {
+    button.addEventListener("click", (e) => {
+      const displayingItem = document.querySelector(".is-displaying");
+      if (e.currentTarget.id === "js-arrow-right-btn") {
+        clickNextButton(displayingItem);
+      } else {
+        clickPrevButton(displayingItem);
+      }
+    });
+  });
+}
+
+const clickPrevButton = (target) => {
+  const targetNextElement = target.previousElementSibling;
+  if (targetNextElement) {
+    target.classList.remove("is-displaying");
+    targetNextElement.classList.add("is-displaying");
+  }
+}
+
+const clickNextButton = (target) => {
+  const targetPrevElement = target.nextElementSibling;
+  if (targetPrevElement) {
+    target.classList.remove("is-displaying");
+    targetPrevElement.classList.add("is-displaying");
+  }
+}
```

Comment:
IMO

You might make common function here. 
Name like `switchImage` etc

```suggestion
const switchImage = (direction) => {
  const displayingItem = document.querySelector(".is-displaying");
  const targetElement = displayingItem[direction];
  if (targetElement) {
    displayingItem.classList.remove("is-displaying");
    targetElement.classList.add("is-displaying");
  }
};
```
The `const displayingItem` part was from `setClickEventInArrowButton`.

After changing↓
```
const setClickEventInArrowButton = () => {
  const arrowButtons = document.querySelectorAll(".arrow-btn");
  arrowButtons.forEach((button) => {
    button.addEventListener("click", (e) => {
     if (e.currentTarget.id === "js-arrow-right-btn") {
        switchImage("nextElementSibling");
      } else {
        switchImage("previousElementSibling");
      }
    });
  });
};
```

Reply:
Thank you very much.
Oh. ....😳 I was able to reduce the number of functions by one.

Please confirm that it has been corrected.
82d7ce1

Reply:
Yay! Looks good to me :) 👍 👍